payments-core/src/main/java/com/stripe/android/networking/AnalyticsRequestFactory.kt

```diff
@@ -21,24 +21,29 @@ internal class AnalyticsRequestFactory @VisibleForTesting internal constructor(
     private val packageManager: PackageManager?,
     private val packageInfo: PackageInfo?,
     private val packageName: String,
-    private val publishableKeyProvider: Provider<String>
+    private val publishableKeyProvider: Provider<String>,
+    private val defaultProductUsageTokens: Set<String> = emptySet()
```

Comment:
Are there cases where we need to set the product token in the function call instead of the constructor specifically, or is the idea that we should over time move everything to the defaultProductUsageToken in the constructor.  

It might be worth adding a comment in here on whichever path you decide and the intent of where it should be headed.

Reply:
We'll still need to keep the method parameters around for products that don't directly make analytics requests. For example, `CardInputWidget` doesn't directly make API calls/analytics requests itself but will pass along product usage details when a `CardParams` instance created by `CardInputWidget` is passed to `Stripe`.

Reply:
atm `AnalyticsRequestFactory` is stateless and can be shared as a singleton across the app. With `defaultProductUsageTokens` added, it seems we're introducing some state to the factory.

Feels like we should only pass `defaultProductUsageTokens` through method parameters in order to make the factory stateless?

Reply:
This is intentional. We should be able to create a scoped `AnalyticsRequestFactory` without every component that calls it needing to know about product usage tokens.